packages/ui/primitives/theme-switcher.tsx

```diff
@@ -0,0 +1,49 @@
+import { motion } from 'framer-motion';
+import { Monitor, MoonStar, Sun } from 'lucide-react';
+import { useTheme } from 'next-themes';
+
+import { useIsMounted } from '@documenso/lib/client-only/hooks/use-is-mounted';
+import { cn } from '@documenso/ui/lib/utils';
+
+export const ThemeSwitcher = () => {
+  const isMounted = useIsMounted();
+
+  const { theme, setTheme } = useTheme();
+
+  const variants = {
+    light: { x: 0 },
+    dark: { x: 'calc((2.15rem + 0.30rem) * 1)' },
+    system: { x: 'calc((2rem + 0.5rem) * 2)' },
```

Comment:
How are these calculated? Maybe we can extract them into variables which can be used as documentation?